### PR TITLE
Add current (server-side) player position highlighting to TileIndicators plugin

### PR DESCRIPTION
**Pull request by pnobbe**
Also add toggles to enable/disable current player position and destination seperately from one another. Same with highlighted tile colors.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsConfig.java [hunk 1]
```diff
@@ -30,19 +30,50 @@
 import net.runelite.client.config.ConfigItem;
 
 @ConfigGroup(
-	keyName = "tileindicators",
-	name = "Tile Indicators",
-	description = "Configuration for the tile indicators plugin"
+		keyName = "tileindicators",
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsConfig.java]: Fix your indentation settings (https://github.com/runelite/runelite/wiki/Code-Conventions). E.g continuation indent should be set to 4 not 8

File: runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsPlugin.java [hunk 2]
```diff
@@ -24,43 +24,42 @@
...
-	@Inject
-	private Client client;
-
-	@Inject
-	private TileIndicatorsConfig config;
-
-	private TileIndicatorsOverlay tileIndicatorsOverlay;
 
 	@Provides
 	TileIndicatorsConfig provideConfig(ConfigManager configManager)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsPlugin.java]: Why you moved method above field declaration?

File: runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsPlugin.java [hunk 3]
```diff
@@ -24,43 +24,42 @@
...
 import net.runelite.client.plugins.PluginDescriptor;
 import net.runelite.client.ui.overlay.Overlay;
 
+import java.util.Arrays;
+import java.util.Collection;
+
 @PluginDescriptor(
-	name = "Tile Indicators",
-	enabledByDefault = false
+		name = "Tile Indicators",
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsPlugin.java]: Fix your indentation

File: runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsPositionOverlay.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import net.runelite.client.ui.overlay.*;
+
+import javax.inject.Inject;
+import java.awt.*;
+
+public class TileIndicatorsPositionOverlay extends Overlay {
+
+	private final Client client;
+	private final TileIndicatorsConfig config;
+
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsPositionOverlay.java]: Remove this empty line

File: runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsPositionOverlay.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+	@Inject
+	TileIndicatorsPositionOverlay(Client client, TileIndicatorsConfig config) {
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsPositionOverlay.java]: Make this constructor private

File: runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsPositionOverlay.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+		setLayer(OverlayLayer.ABOVE_SCENE);
+		setPriority(OverlayPriority.LOW);
+
+		this.client = client;
+		this.config = config;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics) {
+		if (config.enablePosition()) {
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsPositionOverlay.java]: Fix your brackets

File: runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsPositionOverlay.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.tileindicators;
+
+import net.runelite.api.Client;
+import net.runelite.api.Perspective;
+import net.runelite.api.coords.LocalPoint;
+import net.runelite.client.ui.overlay.*;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsPositionOverlay.java]: Fix imports

File: runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsPositionOverlay.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+		setPriority(OverlayPriority.LOW);
+
+		this.client = client;
+		this.config = config;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		if (config.enablePosition())
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsPositionOverlay.java]: Swap this condition with early return

File: runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsDestinationOverlay.java [hunk 9]
```diff
@@ -36,37 +37,43 @@
...
+		this.client = client;
+		this.config = config;
 	}
 
 	@Override
 	public Dimension render(Graphics2D graphics)
 	{
-		LocalPoint dest = client.getLocalDestinationLocation();
-		if (dest == null)
+		if (config.enableDestination())
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsDestinationOverlay.java]: Swap this condition with early return

File: runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsPlugin.java [hunk 10]
```diff
@@ -25,26 +25,31 @@
...
 import javax.inject.Inject;
-import net.runelite.api.Client;
+
+import lombok.extern.slf4j.Slf4j;
 import net.runelite.client.config.ConfigManager;
 import net.runelite.client.plugins.Plugin;
 import net.runelite.client.plugins.PluginDescriptor;
 import net.runelite.client.ui.overlay.Overlay;
 
+import java.util.Arrays;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsPlugin.java]: Fix your imports based on code style guide on wiki

File: runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsDestinationOverlay.java [hunk 11]
```diff
@@ -36,37 +37,43 @@
...
+	private TileIndicatorsDestinationOverlay(Client client, TileIndicatorsConfig config)
 	{
-		this.client = client;
-		this.config = config;
 		setPosition(OverlayPosition.DYNAMIC);
 		setLayer(OverlayLayer.ABOVE_SCENE);
 		setPriority(OverlayPriority.LOW);
+
+		this.client = client;
+		this.config = config;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsDestinationOverlay.java]: Why this change?

File: runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsDestinationOverlay.java [hunk 12]
```diff
@@ -36,37 +37,43 @@
 import net.runelite.client.ui.overlay.OverlayPriority;
 import net.runelite.client.ui.overlay.OverlayUtil;
 
-public class TileIndicatorsOverlay extends Overlay
+import javax.inject.Inject;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsDestinationOverlay.java]: Fix your imports based on wiki

File: runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsDestinationOverlay.java (hunk 11 above)
Comment (pnobbe) [reply runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsDestinationOverlay.java]: For consistency in relation to the PositionOverlay.

File: runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsDestinationOverlay.java [hunk 13]
```diff
@@ -35,24 +35,32 @@
...
+	private TileIndicatorsDestinationOverlay(Client client, TileIndicatorsConfig config)
 	{
-		this.client = client;
-		this.config = config;
 		setPosition(OverlayPosition.DYNAMIC);
 		setLayer(OverlayLayer.ABOVE_SCENE);
 		setPriority(OverlayPriority.LOW);
+
+		this.client = client;
+		this.config = config;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsDestinationOverlay.java]: Don't reformat code for no reason. Also, the variables in constructor should be assigned before any custom properties, so do that. And do same in new overlay too

**Comment by deathbeam** [comment]
Any update on this?

File: runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsPositionOverlay.java [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+import net.runelite.client.ui.overlay.OverlayUtil;
+
+public class TileIndicatorsPositionOverlay extends Overlay
+{
+
+	private final Client client;
+	private final TileIndicatorsConfig config;
+
+	@Inject
+	private TileIndicatorsPositionOverlay(Client client, TileIndicatorsConfig config)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/tileindicators/TileIndicatorsPositionOverlay.java]: This position overlay and destination overlay should probably be merged into 1 overlay and draw both.
